### PR TITLE
fixes setup and some files for consistency

### PR DESCRIPTION
**Pull request by HipsterBrown**
- adds a package.json for easier setup
- renames network-api for consistency
- fixes some links to point to new gitbook paths
- removes Python from supported languages list

pinging @Frijol for review.

### CODE REVIEW

File: API/Languages.md [hunk 1]
```diff
@@ -1,5 +1,4 @@
 # Supported languages
 
-* JavaScript/Node
-* [Python (work in progress)](https://github.com/tessel/tessel-python)
```
Comment (Frijol) [inline API/Languages.md]: Should go in a separate PR

File: API/Modules.md [hunk 2]
```diff
@@ -14,12 +14,12 @@ The best place to find API documentation for a Tessel module is through the link
 
 ## Quick links
 
-* [See all the modules and find their API documentation](//tessel.io/modules)
-* [Learn how to use modules on the start page](//start.tessel.io/modules)
+* [See all the modules and find their API documentation](https://tessel.io/modules)
```
Comment (Frijol) [inline API/Modules.md]: Why force https here? The double slash should keep whatever you currently have (if http, stay http), then I think we force https from our own http site.

File: API/Modules.md [hunk 3]
```diff
@@ -14,12 +14,12 @@ The best place to find API documentation for a Tessel module is through the link
...
+* [Learn how to use modules on the start page](http://tessel.github.io/t2-start/)
```
Comment (Frijol) [inline API/Modules.md]: good catch, but should also go in a separate PR. Would like to isolate changes in this PR to gitbook, not to link & fact fixing

File: README.md [hunk 4]
```diff
@@ -1,16 +1,19 @@
 Developer guide to Tessel 2 Docs Gitbook
 ============
 
-# Tools
-Install the following command line tools using `npm i {{toolname}} -g`
+# Requirements
+The docs are build using a command line tool called [GitBook](https://www.npmjs.com/package/gitbook), which requires:
```
Comment (Frijol) [inline README.md]: s/build/built

File: README.md [hunk 5]
```diff
@@ -1,16 +1,19 @@
 Developer guide to Tessel 2 Docs Gitbook
 ============
 
-# Tools
-Install the following command line tools using `npm i {{toolname}} -g`
```
Comment (Frijol) [inline README.md]: We still need install instructions. In order to use a Node command line tool, you have to install it globally

File: README.md [hunk 6]
```diff
@@ -1,16 +1,19 @@
...
+# Requirements
+The docs are build using a command line tool called [GitBook](https://www.npmjs.com/package/gitbook), which requires:
 
-* [gitbook](https://www.npmjs.com/package/gitbook)
```
Comment (Frijol) [inline README.md]: Because of the need for global install, still need to call these out in the instructions

File: README.md [hunk 7]
```diff
@@ -1,16 +1,19 @@
 Developer guide to Tessel 2 Docs Gitbook
 ============
 
-# Tools
-Install the following command line tools using `npm i {{toolname}} -g`
+# Requirements
```
Comment (Frijol) [inline README.md]: Why the preference? (I'm curious)

File: README.md [hunk 8]
```diff
@@ -1,16 +1,19 @@
...
 
-* [gitbook](https://www.npmjs.com/package/gitbook)
-* [gitbook-summary](https://github.com/imfly/gitbook-summary)
+* [Node Version 4 or higher](https://nodejs.org/en/)
+* [NPM Version 2 or higher](https://docs.npmjs.com/getting-started/installing-node)
 
 # Running locally
 
-Clone, `npm install`, and `gitbook serve`
+* `git clone https://github.com/tessel/t2-docs.git`
```
Comment (Frijol) [inline README.md]: Better to let people choose their own https/ssh preference. I assume this change was made for simpler accessibility, but if so it's a mysterious line of code. Better to link out to github's instructions on cloning.

File: README.md [hunk 9]
```diff
@@ -1,16 +1,19 @@
 Developer guide to Tessel 2 Docs Gitbook
```
Comment (Frijol) [inline README.md]: Note to you/also self: once we get this on gitbooks.io, we can use their simple editor - which doesn't require you to know how to set up a dev environment

File: README.md [hunk 10]
```diff
@@ -1,16 +1,19 @@
...
-* [gitbook](https://www.npmjs.com/package/gitbook)
-* [gitbook-summary](https://github.com/imfly/gitbook-summary)
+* [Node Version 4 or higher](https://nodejs.org/en/)
+* [NPM Version 2 or higher](https://docs.npmjs.com/getting-started/installing-node)
 
 # Running locally
 
-Clone, `npm install`, and `gitbook serve`
+* `git clone https://github.com/tessel/t2-docs.git`
+* `cd t2-docs`
```
Comment (Frijol) [inline README.md]: Add explanation to each line - "enter the folder" or similar

File: README.md [hunk 11]
```diff
@@ -1,16 +1,19 @@
...
+* `npm install`
+* `npm start`
 
 # Changing the contents
 
-SUMMARY.md is autogenerated using the gitbook-summary tool. The command to regenerate is `book sm`. It receives instructions from book.json.
+SUMMARY.md is autogenerated using the gitbook-summary tool, using instructions from `book.json`. The command to regenerate is `npm run build`, which is automatically run during the `npm start` command.
```
Comment (Frijol) [inline README.md]: oh, nicely done!

File: SUMMARY.md [hunk 12]
```diff
@@ -1,15 +1,16 @@
...
-  * [Introduction](/Introduction.md)
+  * [LAN Discovery](debugging/LAN_Discovery.md)
+  * [Repos](debugging/Repos.md)
+  * [USB](debugging/USB.md)
 - Tutorials
-    * [Communication Protocols](/Tutorials/Communication_Protocols.md)
-    * [Making Your Own Module](/Tutorials/Making_Your_Own_Module.md)
+  * [Communication Protocols](Tutorials/Communication_Protocols.md)
+  * [Making Your Own Module](Tutorials/Making_Your_Own_Module.md)
+* [Introduction](Introduction.md)
```
Comment (Frijol) [inline SUMMARY.md]: Still working on this?

File: book.json [hunk 13]
```diff
@@ -5,7 +5,7 @@
         "readme": "Introduction.md"
       },
     "catalog": "all",
-    "ignores": ["_book", ".git"],
+    "ignores": ["_book", "node_modules", "Introduction.md"],
```
Comment (Frijol) [inline book.json]: I don't think we want to ignore the Introduction. What does this do?

File: API/Languages.md (hunk 1 above)
Comment (HipsterBrown) [inline API/Languages.md]: Ok, that makes sense. I'll revert that.

File: API/Modules.md (hunk 3 above)
Comment (HipsterBrown) [inline API/Modules.md]: Makes sense. I'll revert this and make sure a follow up is ready.

File: SUMMARY.md (hunk 12 above)
Comment (HipsterBrown) [inline SUMMARY.md]: Sort of. I think I know how to fix this. Going to try it and push if it works.

File: book.json (hunk 13 above)
Comment (HipsterBrown) [inline book.json]: This was an effort to get `gitbook-summary` to ignore that file since it was replacing the `README` as the intro. This config property only works for directories, so `Introduction.md` still shows up.

File: README.md (hunk 9 above)
Comment (HipsterBrown) [inline README.md]: That's awesome! That will make it a bit more approachable for editorial folks who would like to contribute. :smile:

File: README.md (hunk 5 above)
Comment (HipsterBrown) [inline README.md]: The way I have it setup with the `npm` scripts means no need for to globally install command line dependencies. So scripts like `gitbook serve` as `npm start` run with root privileges. 

This should make it easier for overall setup and reduce clashing global dependency versions because it's all scoped to this project directory.

File: README.md (hunk 6 above)
Comment (HipsterBrown) [inline README.md]: I addressed this [here](https://github.com/tessel/t2-docs/pull/77#discussion_r68756925)

File: README.md (hunk 7 above)
Comment (HipsterBrown) [inline README.md]: I addressed this [here](https://github.com/tessel/t2-docs/pull/77#discussion_r68756925)

File: API/Modules.md (hunk 2 above)
Comment (HipsterBrown) [inline API/Modules.md]: Oh I found the links being generated incorrectly and saw that explicit http or https would fix that. 

screen shot 2016-06-28 at 9 49 13 am

File: SUMMARY.md (hunk 12 above)
Comment (HipsterBrown) [inline SUMMARY.md]: Couldn't fix it. The problem is `gitbook-summary` takes the directory structure and `book.json` to generate the `SUMMARY.md` file, which is ordered alphabetically by directories first and then individual files. Then `gitbook` also uses `book.json` and `SUMMARY.md` to generate the static site, and since the `book.json` configuration tells `gitbook` to replace the README with `Introduction.md`, which is already included in the `SUMMARY.md`, then `Introduction.md` becomes the `index.html` of the site but stays last in the navigation (based on the summary file). 

I thought the solution to this would be to tell `gitbook-summary` to ignore `Introduction.md` but the `ignores` configuration only works on directories at the moment. I may open a PR on that plugin to allow file paths to that configuration, so we don't have this problem anymore.

File: README.md (hunk 5 above)
Comment (Frijol) [inline README.md]: Ah, right! So I see. Cool.

File: SUMMARY.md (hunk 12 above)
Comment (Frijol) [inline SUMMARY.md]: :/ I guess it's not the worst thing.
